### ISSUE TITLE
Support jersey client 2.30

### ISSUE THREAD

**Issue by fujigon** [DataDog/dd-trace-java#1324]
When we use jersey client 2.3.0 with datadog APM, we get exception

```
classCastException: datadog.trace.instrumentation.connection_error.jersey.JerseyClientConnectionErrorInstrumentation$WrappedFuture cannot be cast to java.util.concurrent.CompletableFuture
```

dd-trace-jave currently supports jersey client 2.27, whose related code as following
https://github.com/jersey/jersey/blob/faa809da43538ce31076b50f969b4bd64caa5ac9/core-client/src/main/java/org/glassfish/jersey/client/JerseyInvocation.java#L680
It expects just a `Future` class.

but, jersey client 2.30 now expects `CompletableFuture`(https://github.com/eclipse-ee4j/jersey/blob/472436d3eca7940b05da0c006685e28057b28a33/core-client/src/main/java/org/glassfish/jersey/client/JerseyInvocation.java#L536), then leads to the error.

**Comment by ekennedy247** [comment]
Also noticed this issue when bumping to the [latest spring boot version](https://docs.spring.io/spring-boot/docs/2.3.0.RELEASE/reference/html/appendix-dependency-versions.html), which bumps the jersey client to 2.3.0. 

Since `CompletableFuture` wasn't added until Java 8, not sure what the right approach would be to maintain backwards compatibility with Java 7 (which seems to be desired in CONTRIBUTING.md?).

If the solution is to do something similar to what was done in the [play instrumentation](https://github.com/DataDog/dd-trace-java/tree/fc59355a699d8a652494a8aff735e73724932c24/dd-java-agent/instrumentation/play-2.4/src/main) where we create a separate package for the Java 8 version, I'm happy to open a PR with an attempt.

**Comment by stephenc** [comment]
I'm seeing this issue with Jersey Client 2.32

**Comment by msenmurugan** [comment]
Upgrading the DD java agent version to 0.0.68 version is also not resolving this issue.

**Comment by msenmurugan** [comment]
@randomanderson Can you please prioritize this issue?

**Comment by stephenc** [comment]
https://github.com/DataDog/dd-trace-java/commit/eaf51897cf34d6e83d6bbeec1ceb3d3d91f95e0c#diff-3bacde0cd018060daabbca127f5fef56a14547ba0b1252db4082669cde1ea605R36 might indicate a better trick to have it switch to `CompletableFuture`

**Comment by richardstartin** [comment]
Since there appears to be some demand for this I will start work on this, it will likely be released in the next release or two.

**Comment by richardstartin** [comment]
Sorry, I intended to get this done sooner. This should be fixed by #2251 which will be released in 0.71.0 within the next ten days.
